cmd/minishift/cmd/root.go

```diff
@@ -83,25 +90,31 @@ var RootCmd = &cobra.Command{
 			isAddonInstallRequired bool
 		)
 
-		if !filehelper.Exists(constants.MakeMiniPath("addons")) {
+		constants.MachineName = constants.ProfileName
+		constants.Minipath = constants.GetMinishiftHomeDir()
+		constants.KubeConfigPath = filepath.Join(constants.Minipath, "machines", constants.MachineName+"_kubeconfig")
+
+		minishiftDirs.Home = constants.Minipath
```

Comment:
Why setting all this explicitly. You have a struct now to keep all this. Create a constructor which takes the home directory and let it initialise the rest itself. It is all about encapsulation. You introduced a struct, but don't really make much use of it atm.

Reply:
Sure. Makes sense.